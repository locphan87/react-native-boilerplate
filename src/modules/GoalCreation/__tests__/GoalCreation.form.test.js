import { testSnapshots } from '../../../utils/test.util'
import { InnerForm, handleSubmit } from '../GoalCreation.form.js'

const props = {
  isValid: false,
  isSubmitting: false,
  handleSubmit: jest.fn()
}

describe('module GoalCreation form', () => {
  testSnapshots(InnerForm, [
    {
      props,
      description: 'basic render'
    },
    {
      props: {
        ...props,
        isValid: true
      },
      description: 'when form is valid'
    }
  ])

  describe('handleSubmit', () => {
    test('should call setSubmitting after submit', async () => {
      const bag = {
        props: {
          onSubmit: jest.fn()
        },
        setSubmitting: jest.fn()
      }
      await handleSubmit({ title: 'foo bar' }, bag)
      expect(bag.props.onSubmit).toBeCalledWith({ title: 'foo bar' })
      expect(bag.setSubmitting).toBeCalledWith(false)
    })
  })
})
