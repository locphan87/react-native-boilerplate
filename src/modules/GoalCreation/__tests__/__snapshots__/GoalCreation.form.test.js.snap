// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module GoalCreation form snapshots basic render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <TextInput
    fieldProps={Object {}}
    key="title"
    name="title"
    placeholder="GOALCREATION.FORM.TITLE"
  />
  <TextInput
    fieldProps={Object {}}
    key="start"
    name="start"
    placeholder="GOALCREATION.FORM.START"
  />
  <TextInput
    fieldProps={Object {}}
    key="current"
    name="current"
    placeholder="GOALCREATION.FORM.CURRENT"
  />
  <TextInput
    fieldProps={Object {}}
    key="end"
    name="end"
    placeholder="GOALCREATION.FORM.END"
  />
  <Button
    TouchableComponent={[Function]}
    buttonStyle={
      Object {
        "borderRadius": 3,
      }
    }
    clear={false}
    containerStyle={
      Array [
        Object {
          "marginTop": 10,
        },
        Object {
          "opacity": 0.7,
        },
      ]
    }
    disabled={true}
    iconRight={false}
    loadingProps={
      Object {
        "color": "white",
        "size": "small",
      }
    }
    onPress={[MockFunction]}
    title="goalCreation.form.submit"
  />
</View>
`;

exports[`module GoalCreation form snapshots when form is valid 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <TextInput
    fieldProps={Object {}}
    key="title"
    name="title"
    placeholder="GOALCREATION.FORM.TITLE"
  />
  <TextInput
    fieldProps={Object {}}
    key="start"
    name="start"
    placeholder="GOALCREATION.FORM.START"
  />
  <TextInput
    fieldProps={Object {}}
    key="current"
    name="current"
    placeholder="GOALCREATION.FORM.CURRENT"
  />
  <TextInput
    fieldProps={Object {}}
    key="end"
    name="end"
    placeholder="GOALCREATION.FORM.END"
  />
  <Button
    TouchableComponent={[Function]}
    buttonStyle={
      Object {
        "borderRadius": 3,
      }
    }
    clear={false}
    containerStyle={
      Array [
        Object {
          "marginTop": 10,
        },
        false,
      ]
    }
    disabled={false}
    iconRight={false}
    loadingProps={
      Object {
        "color": "white",
        "size": "small",
      }
    }
    onPress={[MockFunction]}
    title="goalCreation.form.submit"
  />
</View>
`;
